lib/rucio/core/heartbeat.py

```diff
@@ -166,19 +192,28 @@ def live(
                    payload=payload).save(session=session)
 
     # assign thread identifier
-    query = session.query(Heartbeats.hostname,
-                          Heartbeats.pid,
-                          Heartbeats.thread_id)\
-                   .with_hint(Heartbeats, "index(HEARTBEATS HEARTBEATS_PK)", 'oracle')\
-                   .filter(Heartbeats.executable == hash_executable)\
-                   .filter(Heartbeats.updated_at >= datetime.datetime.utcnow() - datetime.timedelta(seconds=older_than))\
-                   .group_by(Heartbeats.hostname,
-                             Heartbeats.pid,
-                             Heartbeats.thread_id)\
-                   .order_by(Heartbeats.hostname,
-                             Heartbeats.pid,
-                             Heartbeats.thread_id)
-    result = query.all()
+    stmt = select(
+        Heartbeats.hostname,
+        Heartbeats.pid,
+        Heartbeats.thread_id
+    ).distinct(
```

Comment:
this `distinct()` was not present in the original query, but this looks correct anyway

Reply:
I think there is basically no difference due to the group_by, but I would still remove it, it just doesn't add anything.

Reply:
Removed